src/quicklist.c

```diff
@@ -1244,25 +1244,26 @@ int quicklistIndex(const quicklist *quicklist, const long long idx,
     initEntry(entry);
     entry->quicklist = quicklist;
 
-    if (!forward) {
-        index = (-idx) - 1;
-        n = quicklist->tail;
-    } else {
-        index = idx;
-        n = quicklist->head;
-    }
-
+    index = forward ? idx : (-idx) - 1;
     if (index >= quicklist->count)
         return 0;
 
+    /* Seek in the other direction if that way is shorter. */
+    int seek_forward = forward;
+    if (index > (quicklist->count - 1) / 2) {
+        seek_forward = !forward;
+        index = quicklist->count - 1 - index;
```

Comment:
Is it possible to add `seek_index`, so that the original `index` is not changed, and so that the `index` is not updated again when the forward is different?

Reply:
OK, I added this in the last commit. Looks better?

Reply:
LGTM.